recipes/angelscript/all/conanfile.py

```diff
@@ -28,68 +32,65 @@ class AngelScriptConan(ConanFile):
     }
 
     short_paths = True
-    exports_sources = "CMakeLists.txt"
-    generators = "cmake"
-    _cmake = None
 
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
-
-    @property
-    def _build_subfolder(self):
-        return "build_subfolder"
-
-    @property
-    def _is_msvc(self):
-        return str(self.settings.compiler) in ["Visual Studio", "msvc"]
+    def export_sources(self):
+        export_conandata_patches(self)
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
+            try:
+                del self.options.fPIC
+            except Exception:
+                pass
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def source(self):
         # Website blocks default user agent string.
-        tools.get(
+        get(
+            self,
             **self.conan_data["sources"][self.version],
-            destination=self._source_subfolder,
+            destination=self.source_folder,
             headers={"User-Agent": "ConanCenter"},
             strip_root=True,
         )
 
-    def _configure_cmake(self):
-        if self._cmake:
-            return self._cmake
-        self._cmake = CMake(self)
-        self._cmake.definitions["AS_NO_EXCEPTIONS"] = self.options.no_exceptions
-        self._cmake.configure(build_folder=self._build_subfolder)
-        return self._cmake
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["AS_NO_EXCEPTIONS"] = self.options.no_exceptions
+        if not valid_min_cppstd(self, 11):
+            tc.variables["CMAKE_CXX_STANDARD"] = 11
```

Comment:
No validate method? it's not in the diff 🤔

Reply:
It's another case here. The project needs C++11 but it's not listed on CMakeLists.txt. Probably the author runs some newer compiler version which uses C++11 or higher b default.